rfcs/0024-wallet-internalization.md

```diff
@@ -0,0 +1,102 @@
+## Summary
+
+---
+
+Integrate an internationalization library into the Coda wallet for multi-language support.
+
+## Motivation
+
+---
+
+The Coda wallet currently has only English text, making it more difficult for non-English speaking developers and users to easily contribute and use.
+
+The wallet should support multiple languages to reach a wide-ranging audience.
+
+## Detailed design
+
+---
+
+React-intl is currently one of the most well-supported internationalization library for React. It is built upon the native **[Internationalization API](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl)** browser API, and neatly wraps them for performance and usability enhancements.
+
+Lib: [https://github.com/formatjs/react-intl](https://github.com/formatjs/react-intl)
+
+Bs-bindings: [https://github.com/reasonml-community/bs-react-intl](https://github.com/reasonml-community/bs-react-intl)
+
+General steps to implement react-intl
+
+1. Create a JSON of IDs mapped to their text. Each new language will have their own JSON file.
+
+Discussion: Am I converting _all_ texts in this grant? It's within capacity. Just not sure if the Coda team is comfortable with having an app-wide change immediately due to some potential internal conflicts.
+
+```json
+    // en.json
+
+    [
+      {
+        "id": "page.hello",
+        "defaultMessage": "Hello",
+        "message": ""
+      }
+    ]
+```
+
+2.  Wrap the app with the react-intl provider.
+
+```javascript
+    // ReactApp.re
+    let make = () => {
+    let settingsValue = AddressBookProvider.createContext();
+    let onboardingValue = OnboardingProvider.createContext();
+
+        // Create the context for react-intl
+
+
+        <ReactIntl.IntlProvider
+        locale={/* Logic to get the right locale */}
+        messages={/* Logic to get the right set of JSON messages  */}>
```

Comment:
Any thoughts on what this logic looks like?
If I had to make something up, I'd guess any user-set locale would be saved in localStorage or something, and maybe we can try to detect with some fallback otherwise? Idk. Presumably we would need to add something to the settingsto be able to change it (maybe a simple dropdown for now if that makes sense).

Would you use some sort of webpack loader to include the json files for all languages in the shipped app or load json from disk at runtime?

Maybe doesn't need to be completely ironed out before starting, just want to start considering these things.

Reply:
I'll add comments and code to showcase the local selection as well as the message selection. 

I've actually only ever included the i8n json files directly in the bundle, albeit in vendor bundles if I'm code-splitting. This is my first time working with Electron, so I don't know how loading assets work with the bundled app, but if there's an optional to load only the right json file, let's do that!

Reply:
We shouldn't have any issue loading json files from disk in electron, will just need to wire something up but I can help with that. Mostly just looking to iron out the big picture.